lib/src/ast_extensions.dart

```diff
@@ -125,6 +125,25 @@ extension AstIterableExtensions on Iterable<AstNode> {
   /// but allow a split if there are elements or comments inside.
   bool canSplit(Token rightBracket) =>
       isNotEmpty || rightBracket.precedingComments != null;
+
+  /// Returns `true` if the collection containing these elements and terminated
+  /// by [rightBracket] contains any line comments before, between, or after
+  /// any elements.
+  ///
+  /// Comments within an element are ignored.
+  bool containsLineComments([Token? rightBracket]) {
+    // Look before each element.
+    for (var element in this) {
+      if (element.beginToken.hasLineCommentBefore) return true;
+    }
+
+    // Look before the closing bracket.
+    if (rightBracket != null) {
+      if (rightBracket.hasLineCommentBefore) return true;
+    }
+
+    return false;
+  }
```

Comment:
[optional] I think the code-golfed version is pretty readable because it looks declarative.

```suggestion
  bool containsLineComments([Token? rightBracket]) =>
      any((element) => element.beginToken.hasLineCommentBefore) ||
      (rightBracket?.hasLineCommentBefore ?? false);
```